libcxx/include/inplace_vector

```diff
@@ -0,0 +1,1091 @@
+
+  static constexpr _LIBCPP_HIDE_FROM_ABI bool __do_destroy() noexcept {
+    if consteval {
+      // Do not destroy elements of the array in a constant expression
+      return !__base::__constexpr;
+    }
+    return true;
+  }
+
+public:
+  // types
+  using value_type      = _Tp;
+  using pointer         = _Tp*;
+  using const_pointer   = const _Tp*;
+  using reference       = value_type&;
+  using const_reference = const value_type&;
+  using size_type       = size_t;
+  using difference_type = ptrdiff_t;
+#  ifdef _LIBCPP_ABI_BOUNDED_ITERATORS
+  using iterator       = __bounded_iter<__wrap_iter<pointer>>;
+  using const_iterator = __bounded_iter<__wrap_iter<const_pointer>>;
+#  else
+  using iterator       = __wrap_iter<pointer>;
+  using const_iterator = __wrap_iter<const_pointer>;
+#  endif
+  using reverse_iterator       = std::reverse_iterator<iterator>;
+  using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+
+  using __trivially_relocatable = __base::__trivially_relocatable;
+
+  // [inplace.vector.cons], construct/copy/destroy
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector() noexcept = default;
+  constexpr _LIBCPP_HIDE_FROM_ABI explicit inplace_vector(size_type __n) : inplace_vector() {
+    if (__n > capacity())
+      __throw_bad_alloc();
+    // TODO: investigate optimization for trivially constructible _Tp that,
+    // for small _Capacity, unconditionally value_construct all the elements to turn
+    // "memset(data(), 0, sizeof(_Tp) * __n)" into
+    // "memset(data(), 0, sizeof(_Tp[_Capacity]))" which might optimize better
+    for (size_type __i = 0; __i < __n; ++__i) {
+      unchecked_emplace_back();
+      _LIBCPP_ASSERT_SEMANTIC_REQUIREMENT(__i + 1zu == __sz_, "inplace_vector modified before its lifetime began");
+    }
+  }
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector(size_type __n, const _Tp& __value) : inplace_vector() {
+    if (__n > capacity())
+      __throw_bad_alloc();
+    for (size_type __i = 0; __i < __n; ++__i) {
+      unchecked_emplace_back(__value);
+      _LIBCPP_ASSERT_SEMANTIC_REQUIREMENT(__i + 1zu == __sz_, "inplace_vector modified before its lifetime began");
+    }
+  }
+
+  // NOTE: [sequence.reqmts]p(67.1) does not apply to this constructor but apply
+  // it anyways so the (size_type, const T&) constructor can never be ambiguous
+  template <class _InputIterator>
+    requires(__has_input_iterator_category<_InputIterator>::value)
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector(_InputIterator __first, _InputIterator __last) : inplace_vector() {
+    if constexpr (__has_forward_iterator_category<_InputIterator>::value) {
+      __append_with_size(__first, __last, std::distance(__first, __last));
+      return;
+    }
+    __append_with_sentinel(__first, __last);
+  }
+  template <_ContainerCompatibleRange<_Tp> _Range>
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector(from_range_t, _Range&& __rg) : inplace_vector() {
+    append_range<true>(__rg);
```

Comment:
```suggestion
    append_range(std::forward<_Range>(__rg));
```
This line does not compile on my machine. I guess this was supposed to forward to the `append_range` member function?